inject-test/src/main/java/io/avaje/inject/test/MetaReader.java

```diff
@@ -349,8 +352,29 @@ Object get(Object instance) {
     }
 
     void setFromScope(BeanScope beanScope, Object testInstance) throws IllegalAccessException {
+      if (valueAlreadyProvided) {
+        return;
+      }
+      final var type = type();
+
+      if (type instanceof ParameterizedType) {
+        final var parameterizedType = (ParameterizedType) type;
+        final var rawType = parameterizedType.getRawType();
+        final var typeArguments = parameterizedType.getActualTypeArguments();
+
+        if (rawType.equals(List.class)) {
+          set(field, beanScope.list(typeArguments[0]), testInstance);
+          return;
+        }
+
+        if (rawType.equals(Optional.class)) {
+          set(field, beanScope.getOptional(typeArguments[0], name), testInstance);
+          return;
+        }
+      }
+
       if (!valueAlreadyProvided) {
```

Comment:
Can this `if` get removed now?  ... cause it's checked with early return right at the top now